src/js/plugin.js

```diff
@@ -0,0 +1,281 @@
+   *         - `name`: The name of the plugin.
+   *         - `plugin`: The plugin class/constructor.
+   *
+   * @return {Boolean}
+   *         Whether or not default was prevented.
+   */
+  trigger(event, hash = {}) {
+    return Events.trigger(this.eventBusEl_, event, this.getEventHash_(hash));
+  }
+
+  /**
+   * Disposes a plugin.
+   *
+   * Subclasses can override this if they want, but for the sake of safety,
+   * it's probably best to subscribe to one of the disposal events.
+   */
+  dispose() {
+    const {name, player} = this;
+
+    this.trigger('dispose');
+    this.off();
+
+    // Eliminate any possible sources of leaking memory by clearing up references
+    // between the player and the plugin instance and nulling out the plugin's
+    // state and replacing methods with a function that throws.
+    player.plugins_[name] = false;
+    this.player = this.state = null;
+
+    this.dispose = () => {
+      throw new Error('cannot call methods on a disposed object');
+    };
+
+    this.setState = this.off = this.on = this.one = this.trigger = this.dispose;
+
+    // Finally, replace the plugin name on the player with a new factory
+    // function, so that the plugin is ready to be set up again.
+    player[name] = createPluginFactory(name, pluginStorage[name]);
+  }
+
+  /**
+   * Determines if a plugin is a "basic" plugin (i.e. not a sub-class of `Plugin`).
+   *
+   * @param  {String|Function} plugin
+   *         If a string, matches the name of a plugin. If a function, will be
+   *         tested directly.
+   * @return {Boolean}
+   */
+  static isBasic(plugin) {
+    plugin = (typeof plugin === 'string') ? Plugin.getPlugin(plugin) : plugin;
+
+    return typeof plugin === 'function' &&
+      !Plugin.prototype.isPrototypeOf(plugin.prototype);
+  }
+
+  /**
+   * Register a Video.js plugin
+   *
+   * @param  {String} name
+   * @param  {Function} plugin
+   *         A sub-class of `Plugin` or an anonymous function for basic plugins.
+   * @return {Function}
+   */
+  static registerPlugin(name, plugin) {
+    if (typeof name !== 'string' || pluginStorage[name] || Player.prototype[name]) {
+      throw new Error(`illegal plugin name, "${name}"`);
+    }
+
+    if (typeof plugin !== 'function') {
+      throw new Error(`illegal plugin for "${name}", must be a function, was ${typeof plugin}`);
+    }
+
+    pluginStorage[name] = plugin;
+
+    if (Plugin.isBasic(plugin)) {
+      Player.prototype[name] = createBasicPlugin(name, plugin);
+    } else {
+      Player.prototype[name] = createPluginFactory(name, plugin);
+    }
+
+    return plugin;
+  }
+
+  /**
+   * Register multiple plugins via an object where the keys are plugin names
+   * and the values are sub-classes of `Plugin` or anonymous functions for
+   * basic plugins.
+   *
+   * @param  {Object} plugins
+   * @return {Object}
+   *         An object containing plugins that were added.
+   */
+  static registerPlugins(plugins) {
```

Comment:
This doesn't seem to add much for end users that they couldn't achieve by repeating `registerPlugin()` a couple times in a row.